frappe/public/js/frappe/form/controls/duration.js

```diff
@@ -123,33 +114,14 @@ frappe.ui.form.ControlDuration = class ControlDuration extends frappe.ui.form.Co
 	}
 
 	get_duration() {
-		// returns an object of days, hours, minutes and seconds from the inputs array
-		let total_duration = {
-			minutes: 0,
-			hours: 0,
-			days: 0,
-			seconds: 0,
-		};
-		if (this.inputs) {
-			total_duration.minutes = parseInt(this.inputs.minutes.val());
-			total_duration.hours = parseInt(this.inputs.hours.val());
-			if (!this.duration_options.hide_days) {
-				total_duration.days = parseInt(this.inputs.days.val());
-			}
-			if (!this.duration_options.hide_seconds) {
-				total_duration.seconds = parseInt(this.inputs.seconds.val());
-			}
+		let total_duration = {};
+		for (const [label, $input] of Object.entries(this.inputs)) {
+			total_duration[label] = parseInt($input.val());
 		}
 		return total_duration;
 	}
 
 	is_duration_picker_set(inputs) {
-		let is_set = false;
-		Object.values(inputs).forEach((duration) => {
-			if (duration.prop("value") != 0) {
-				is_set = true;
-			}
-		});
-		return is_set;
+		return Object.values(inputs).some((input) => parseInt(input.val()) !== 0);
 	}
```

Comment:
While using “some“ makes it slightly more efficient and parseInt() should make it a bit more reliable, this is mere refactoring.